x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts

```diff
@@ -0,0 +1,240 @@
+    getMaxQueued: jest.fn().mockReturnValue(100),
+  };
+
+  const mockLogger: Logger = {
+    trace: jest.fn(),
+    debug: jest.fn(),
+    info: jest.fn(),
+    warn: jest.fn(),
+    error: jest.fn(),
+    fatal: jest.fn(),
+    log: jest.fn(),
+    isLevelEnabled: jest.fn().mockReturnValue(true), // Enable all levels by default
+    get: jest.fn((...childContextPaths: string[]) => mockLogger), // Return the same mock for child loggers
+  };
+
+  const mockServices: Services = {
+    savedObjectsClient: {} as SavedObjectsClientContract, // Empty mock object
+    scopedClusterClient: {} as ElasticsearchClient, // Empty mock object
+    connectorTokenClient: {} as ConnectorTokenClient, // Empty mock object
+  };
+
+  const mockHeaders: Headers = {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer some-token',
+    'X-Custom-Header': ['value1', 'value2'], // Multiple values
+  };
+
+  const mockAxiosHeaders = new AxiosHeaders({
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer some-token',
+    'X-Custom-Header': ['value1', 'value2'],
+  });
+
+  const mockKibanaRequest: KibanaRequest = {
+    id: 'mock-request-id',
+    uuid: 'mock-request-uuid',
+    url: new URL('https://example.com/api/endpoint'),
+    route: {
+      path: '/api/endpoint',
+      method: 'get',
+      options: {
+        authRequired: 'optional',
+        xsrfRequired: false,
+        access: 'public',
+        tags: ['readonly'],
+        timeout: {
+          payload: undefined,
+          idleSocket: undefined,
+        },
+        description: 'string',
+      },
+    },
+    headers: mockHeaders,
+    isSystemRequest: false,
+    isFakeRequest: false,
+    isInternalApiRequest: false,
+    socket: {} as IKibanaSocket, // You might need to mock this depending on your code's interaction with it
+    events: {} as KibanaRequestEvents, // You might need to mock this too
+    auth: {
+      isAuthenticated: false,
+    },
+    params: {},
+    query: {},
+    body: {},
+  };
+
+  beforeEach(() => {
+    mockServiceParams = {
+      connector: { id: 'Gemini-id', type: 'gemini' },
+      configurationUtilities: mockConfigurationUtilities,
+      config: {
+        apiUrl: 'https://api.gemini.com',
+        defaultModel: 'gemini-1.5-pro-preview-0409',
+        gcpRegion: 'us-central1',
+        gcpProjectID: 'my-project-12345',
+      },
+      secrets: {
+        credentialsJson: 'some-random-string',
+      },
+      logger: mockLogger,
+      services: mockServices,
+      request: mockKibanaRequest,
+    };
+
+    connector = new GeminiConnector(mockServiceParams);
+  });
+
+  describe('getDashboard', () => {
+    it('should return available: true when user has privileges', async () => {
+      const dashboardId = 'some-dashboard-id';
+
+      // Mock initDashboard to simulate a successful dashboard fetch
+      jest.spyOn(connector, 'initDashboard').mockResolvedValueOnce({ success: true });
```

Comment:
I see an error here: `Argument of type "initDashboard" is not assignable to parameter of type "getDashboard" | "runApi" | "getSubActions"`

`initDashboard` is a function imported into the file. By mocking the import, you should be able to set a mock on this function. You can follow [how it is done in the bedrock test file](https://github.com/elastic/kibana/blob/0fde2774a68cd449529ee615a3d46132c2097715/x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.test.ts#L28).

Reply:
cc: @shivankawasthi11

Reply:
@stephmilovic Reworked on the implementation taking reference from the example provided